### ISSUE TITLE
libswift_Concurrency.dylib not found on macOS 10.15 & macOS 11

### ISSUE THREAD

**Issue by miketimofeev** [realm/SwiftLint#4131]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Invoking `swiftlint version` on 0.49.0 produces an error:
`dyld: Library not loaded: /usr/lib/swift/libswift_Concurrency.dylib   Referenced from: /usr/local/bin/swiftlint   Reason: image not found
` on macOS 10.15 and macOS 11. macOS 12 works fine though.

##### Complete output when running SwiftLint, including the stack trace and command used

n\a

### Environment

We install swiftling using portable package via the installation script https://github.com/actions/runner-images/blob/main/images/macos/provision/core/swiftlint.sh

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
let foo = try! bar()
```

**Comment by miketimofeev** [comment]
@jpsim ugh, just noticed that breaking change
![image](https://user-images.githubusercontent.com/48208649/187020407-cb74ef64-3731-40fd-a6fa-8d38c5193c4e.png)
Does it mean we should stick to 0.48.0 for macOS 10.15 & macOS 11?

**Comment by jpsim** [comment]
That’s right, this was an intended breaking change in order to allow us to use swift concurrency features.

**Comment by jpsim** [comment]
We occasionally have “release candidate” releases like we did here for 0.49.0 (rc1 & rc2) to help users identify if there’s a breaking change or regression they need to flag with us or account for.

**Comment by miketimofeev** [comment]
Thanks for the suggestion but we try to avoid installing RC versions of tools to our images.

**Comment by jpsim** [comment]
I’m not saying you deploy RC changes, I’m saying we make RCs so people can proactively catch issues before they ship. How can I help you to do that?

**Comment by miketimofeev** [comment]
I'm afraid our tests during the VM image generation cover only basic scenarios like version output so in most cases we won't be able to test the RC versions properly without providing the image with such new versions to customers.
